app.js

```diff
@@ -419,6 +419,44 @@ function ApiHandler(compileHandler) {
             }).join("\n"));
         }
     }, this));
+     // ---- this whole thing should be in a separate file etc, proof of concept only
+    // TODO: we'll need something smarter than this; URLs are long-lived and we nede
+    // to think about it; languages...formatters...etc.maybe something like:
+    var exec = require('./lib/exec');
+    this.handler.post('/format/clang-format-3.8', function (req, res) {
```

Comment:
This needs to be made configurable like the compilers. `clang-format` only works if clang is installed in the path: on the CE site I have so many compilers and none are in the path: I must refer to them directly by full path (see all the entries in `c++.amazon.properties` for example)